xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/XWikiVersioningStoreInterface.java

```diff
@@ -47,6 +55,79 @@ void saveXWikiDocArchive(XWikiDocumentArchive archivedoc, boolean bTransaction,
 
     Version[] getXWikiDocVersions(XWikiDocument doc, XWikiContext context) throws XWikiException;
 
+    /**
+     * Gets versions from a given document matching criteria like author, minimum creation date, etc.
+     *
+     * @param doc the document
+     * @param criteria criteria used to match versions
+     * @param context the XWiki context
+     * @return the matching versions
+     * @since 15.10.7
+     * @since 16.2.0RC1
+     */
+    @Unstable
+    default Collection<Version> getXWikiDocVersions(XWikiDocument doc, RevisionCriteria criteria, XWikiContext context)
```

Comment:
Also, wouldn't it be interesting to refactor a little this method to call `stream().count()` [here](https://github.com/xwiki/xwiki-platform/pull/2925/files#diff-7675da62f39751ae6a949342a3ef04af1d629d3c9bcc39313fb955dd1093ad91R111), allowing to avoid aggregating the steam to a list when we just want a count (e.g., [here](https://github.com/xwiki/xwiki-platform/pull/2925/files#diff-7675da62f39751ae6a949342a3ef04af1d629d3c9bcc39313fb955dd1093ad91R128)).
It's a small perf improvement, but since we are focusing on versions perfs here.